silhouette-password-bcrypt/src/main/scala/com/mohiva/play/silhouette/password/BCryptPasswordHasher.scala

```diff
@@ -51,7 +47,11 @@ case class BCryptPasswordHasher(logRounds: Int = 10) extends PasswordHasher {
    * @param plainPassword The password to hash.
    * @return A PasswordInfo containing the hashed password.
    */
-  override def hash(plainPassword: String) = PasswordInfo(id, BCrypt.hashpw(plainPassword, BCrypt.gensalt(logRounds)))
+  override def hash(plainPassword: String) = PasswordInfo(
+    hasher = id,
+    password = BCrypt.hashpw(plainPassword, BCrypt.gensalt(logRounds)),
+    metadata = Some(logRounds.toString)
```

Comment:
The metadata is actually more complicated because it should also include the version of the bcrypt algorithm. bcrypt hashes follow a [modular crypt format](https://pythonhosted.org/passlib/modular_crypt_format.html):

> In short, the modular crypt format (MCF) is a standard for encoding password hash strings, which requires hashes have the format `$identifier$content`; where `identifier` is an short alphanumeric string uniquely identifying a particular scheme, and `content` is the contents of the scheme, using only the characters in the regexp range `[a-zA-Z0-9./]`.

Currently `identifier = "2a$" + logRounds`. The `2a` part can be changed in the future if the algorithm is updated, so that should be stored. I think a decent future-proof method would be `metadata = password.substring(1, password.lastIndexOf('$'))`.

Note the redundancy here: metadata is computed from the password field! This is why I think it's odd that you want to add a column. `BCryptPasswordHasher` is perfectly capable of implementing `isDeprecated` and `isSuitable` based on `PasswordInfo` as it is now without `metadata` and without even needing to change `id`. Really it would take quite an awful hasher implementation to be able to implement `matches` correctly but not `isSuitable` and `isDeprecated` based only on the current `PasswordInfo`. Between `id` and `password` I think there is plenty of space to store as much metadata as needed without causing any inconvenience. Currently `password` is comfortably storing 4 pieces of data in a single string for bcrypt.